pkg/resources/function.go

```diff
@@ -61,9 +65,11 @@ var functionSchema = map[string]*schema.Schema{
 		Type:        schema.TypeString,
 		Description: "The return type of the function",
 		// Suppress the diff shown if the values are equal when both compared in lower case.
-		DiffSuppressFunc: DiffTypes,
-		Required:         true,
-		ForceNew:         true,
+		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+			return strings.EqualFold(old, new)
+		},
+		Required: true,
+		ForceNew: true,
```

Comment:
This is the behavior change, please add a note to the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#migration-guide)

Reply:
added note to migration guide